### PR TITLE
[Redesign] Assume dates are UTC (even though their DateTimeKind is Unspecified)

### PR DESCRIPTION
**Pull request by joelverhagen**
Also, never show relative years -- show full dates at that point

Fix https://github.com/NuGet/NuGetGallery/issues/4280
Fix https://github.com/NuGet/NuGetGallery/issues/4424

### CODE REVIEW

File: src/NuGetGallery/Scripts/gallery/common.js [hunk 1]
```diff
@@ -200,7 +200,7 @@
 
         var allButLast = items.slice(0, -1).join(', ');
         var last = items.slice(-1)[0];
-        return [allButLast, last].join(items.length < 2 ? '' : (items.length == 2 ? ' and ' : ', and '));
+        return [allButLast, last].join(items.length < 2 ? '' : items.length === 2 ? ' and ' : ', and ');
```
Comment (skofman1) [inline src/NuGetGallery/Scripts/gallery/common.js]: === ?

Comment (joelverhagen) [inline src/NuGetGallery/Scripts/gallery/common.js]: `===` matches type in addition to what `==` does. Equal value and equal type. It's a best practice.

File: src/NuGetGallery/Services/TelemetryService.cs [hunk 2]
```diff
@@ -126,13 +126,13 @@ private static string GetClientVersion()
 
         private static string GetAccountCreationDate(User user)
         {
-            return user.CreatedUtc != null ? user.CreatedUtc.Value.ToString("o") : "N/A";
+            return user.CreatedUtc != null ? user.CreatedUtc.Value.ToString("O") : "N/A";
```
Comment (loic-sharma) [inline src/NuGetGallery/Services/TelemetryService.cs]: You can use `user.CreatedUtc?.Value.ToString("O")  ?? "N/A"` to be consistent with line 135.

Comment (joelverhagen) [inline src/NuGetGallery/Services/TelemetryService.cs]: Good idea.

Comment (joelverhagen) [inline src/NuGetGallery/Services/TelemetryService.cs]: Fixed.

**Pull request merged**